src/Mod/Part/BasicShapes/SkipGui.py

```diff
@@ -0,0 +1,250 @@
+#! python
+# -*- coding: utf-8 -*-
+# SkipGui.py
+# 2022, by Mark Ganson <TheMarkster>
+# LGPL 2.1 or later
+
+# this file is called from c++ in Mod/Part/Gui/Command.cpp
+# from various functions when these commands are activated:
+#   Part::Loft
+#   Part::Sweep
+#   Part::Extrude
+#   Part::Revolve
+# Then if the user has preselected certain objects and subobjects
+# the Gui can be skipped and the document objects directly created
+
+# If False is returned, then the Gui dialog should be shown
+# Else if True is returned, then the dialog can be skipped
+
+import FreeCAD as App
+import FreeCADGui as Gui
+
+def getActStr(document):
+    return "App.getDocument(\'" + document.Name + "\').ActiveObject"
+
+def getDocStr(document):
+    return "App.getDocument(\'" + document.Name +"\')"
+
+def getObjStr(document, object):
+    return getDocStr(document) + ".getObject(\'" + object.Name + "\')"
+
+def getVisStr(document, objs):
+    ret = ""
+    for obj in objs:
+        ret += getDocStr(document) + ".getObject(\'" + obj.Name + "\').ViewObject.Visibility = False\n"
+    return ret
+
+def getSetPropListStr(document, propName, objList):
+    ret = getActStr(document) + "." + propName + " = ["
+    for obj in objList:
+        ret += getDocStr(document) + ".getObject(\'" + obj.Name + "\'), "
+    ret += "]\n"
+    return ret
+
+def getAddObjStr(document, typeId, name):
+    ret = getDocStr(document)
+    ret += ".addObject(\'" + typeId + "\', \'" + name + "\')\n"
+    return ret
+
+def getSubListStr(names):
+    ret = "["
+    for name in names:
+        ret += "\'" + name + "\', "
+    ret += "]"
+    return ret
+
+
+def makeExtrude():
+
+    selx = Gui.Selection.getSelectionEx()
+    if len(selx) == 0:
+        App.Console.PrintMessage(f"makeExtrude() len(selx) = {len(selx)} (must be > 0 to skip dialog)\n")
+        return False
+
+    selected = [sel.Object for sel in selx if bool(sel.Object.isDerivedFrom("Part::2DObject") or
+                sel.Object.isDerivedFrom("Part::FeaturePython") or
+                bool(hasattr(sel.Object,"Shape") and len(sel.Object.Shape.Vertexes) == 1) or
+                sel.Object.isDerivedFrom("Sketcher::SketchObject")) and
+                sel.HasSubObjects == False]
+    if len(selected) != 1:
+        App.Console.PrintMessage(f"makeExtrude() len(selected) = {len(selected)} (must be 1 to skip dialog)\n")
+        return False #need exactly 1 profile for Extrude
+
+    solid = "True"
+    base = selected[0]
+    if hasattr(base.Shape, "Wire1"):
+        if not base.Shape.Wire1.isClosed():
+            solid = "False"
+    else:
+        if not bool(hasattr(base,"Shape") and len(base.Shape.Vertexes) == 1):
+            return False #no Wire1
+        else:
+            solid = "False"
+
+    spines = [sel for sel in selx if sel.HasSubObjects]
```

Comment:
Don't much understand? If all edges are from same object, `getSelectionEx` will return only one object.

Reply:
Yes, getSelectionEx() only returns 1 object if all edges are from the same object, but we don't know that to be the case yet.  The user might have selected multiple edges from multiple objects, in which case we'll not skip the dialog.

HasSubObjects ensures these objects were selected in the 3d view and not in the tree.  For extrude this will be used for the DirLink property which is of type App::PropertyLinkSub, which only accepts a single object.  It would only be sensible for 1 object with this property anyway since it defines the direction and distance to extrude, but for Sweep it could be sensible to use App::PropertyLinkSubList instead, to support edges from multiple objects to define the path.  But I stray off topic.

So we need to check here that len(spines) = 1 so we can use spines[0] in the next bit of code.